src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBufferReader.cs

```diff
@@ -231,7 +231,7 @@ private bool TryEnsureBytes(int count)
                 }
                 int needed = newOffsetMax - _offsetMax;
                 DiagnosticUtility.DebugAssert(needed > 0, "");
-                int read = _stream.ReadAtLeast(_buffer.AsSpan(_offsetMax, needed), needed, throwOnEndOfStream: false);
+                int read = _stream.ReadAtLeast(_buffer.AsSpan(_offsetMax), needed, throwOnEndOfStream: false);
```

Comment:
Prior to the change to use ReadAtLeast, this was also limiting how much was read to `needed`.  Is it possible changing it to read more could create bugs?  For example, is it possible this might end up reading more from the stream that a consumer intended, such that it would erroneously consume something from the stream _after_ what is being deserialized?

Reply:
Good catch,  I am thinking about closing this for now and maybe add a comment in a separate PR.

Is there any other reader which have any way around that (such as resetting the stream position) ?

Reply:
Resetting position isn't always possible: many (most) streams don't support it.

We should just close this PR for now.  Thanks, though.